tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -44,158 +54,280 @@ public static void OnInstrumentPublished(Instrument instrument, MeterListener li
+            {
+                return InstrumentType.UpDownCounter;
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableUpDownCounter`1"))
+            {
+                return InstrumentType.ObservableUpDownCounter;
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Gauge`1"))
+            {
+                return InstrumentType.Gauge;
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableGauge`1"))
+            {
+                return InstrumentType.ObservableGauge;
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Histogram`1"))
+            {
+                return InstrumentType.Histogram;
             }
-        }
 
-        internal static void TriggerAsyncCollection()
-        {
-            Log.Debug("Manually triggering async instrument collection...");
-            MetricReader.CollectObservableInstruments();
-            Log.Debug("Async collection triggered.");
+            return null;
         }
 
-        internal static IReadOnlyDictionary<string, MetricPoint> GetMetricsForExport(bool clearAfterGet = false)
+        private static MetricState? CreateMetricState(Instrument instrument)
         {
-            var metrics = CapturedMetrics;
-
-            if (clearAfterGet)
+            try
             {
-                CapturedMetrics.Clear();
-            }
+                var instrumentType = instrument.GetType().FullName;
+                if (instrumentType is null)
+                {
+                    Log.Warning("Unable to get the full name of the instrument type for: {InstrumentName}", instrument.Name);
+                    return null;
+                }
 
-            return metrics;
-        }
+                var aggregationType = GetInstrumentType(instrumentType);
+                if (aggregationType == null)
+                {
+                    Log.Debug("Skipping unsupported instrument: {InstrumentName} of type: {InstrumentType}", instrument.Name, instrumentType);
+                    return null;
+                }
 
-        // For testing only
-        internal static IReadOnlyDictionary<string, MetricPoint> GetCapturedMetricsForTesting() => CapturedMetrics;
+                var metricStreamIdentity = new MetricStreamIdentity(instrument, aggregationType.Value);
+                var temporality = GetTemporalityString(aggregationType.Value);
+                var tagsDict = new Dictionary<string, object?>();
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static string GetTemporalityString()
-        {
-            return _cachedTemporality ??= Tracer.Instance.Settings.OtlpMetricsTemporalityPreference.ToString();
-        }
+                // RFC requirement: Check for duplicate instrument registration
+                if (CapturedMetrics.Any(kvp =>
```

Comment:
It would be worth checking to see if the runtime already handles this so we don't have to. It would be awesome if the runtime already did checks so when an instrument is created twice with the same information, it returns the _same_ instrument object (`Object.Equals(instrument1, instrument2) == true`)